handlers/app.go

```diff
@@ -0,0 +1,48 @@
+package handlers
+
+
+import (
+
+	"github.com/knative-scout/app-api/models"
+	"go.mongodb.org/mongo-driver/bson"
+	"net/http"
+	"github.com/gorilla/mux"
+	"fmt"
+)
+
+// AppByIDHandler returns a single app by ID from the database
+type AppByIDHandler struct {
+	BaseHandler
+}
+// ServeHTTP implements http.Handler
+func (h AppByIDHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	//gets all the optional parametres passed in the URL
+	vars := mux.Vars(r)
+
+	id := vars["id"]
+
+	resp := getDataFromDB(id, h)
+
+	h.RespondJSON(w, http.StatusOK, resp)
+}
+
+
+func getDataFromDB(id string, h AppByIDHandler ) models.App{
```

Comment:
If you go to `localhost:5000/apps/this-app-id-does-not-exist` your program will crash.

Can you make it so the endpoint responds with `http.StatusNotFound` and a response of `{"error": "not found"}`?